### PR TITLE
[CHARLIE-NO-TICKET] X-Tracking-ID support

### PR DESCRIPTION
**Pull request by RaVbaker**
To support new `X-Tracing-Id` header as main source of truth for main trace we deprecating the `X-Request-Id`. If provided it will seed the `X-Tracking-Id` and be added to all subsequent calls in downstream. Actually for backwards compatibility both headers will be with exactly same value and should match. 

For full support I had to introduce GitHubActions and adjust specs so they will pass in new environment (with more up2date Rails and Ruby). Because our current projects uses more recent version of Ruby than 2.3 and Rails 4.1. 

Since travis is deprecated for [open source projects](https://www.jeffgeerling.com/blog/2020/travis-cis-new-pricing-plan-threw-wrench-my-open-source-works) I moved the Checks to GitHub Actions.

### CODE REVIEW

File: lib/kiev/context_reader.rb [hunk 1]
```diff
@@ -6,23 +6,28 @@ module Kiev
   # change field lookup.
   class ContextReader
     REQUEST_ID = "request_id"
+    TRACKING_ID = "tracking_id"
     REQUEST_DEPTH = "request_depth"
     TREE_PATH = "tree_path"
 
     def initialize(subject)
+      subject[TRACKING_ID] ||= subject[REQUEST_ID]
```
Comment (giulliano-bueno) [inline lib/kiev/context_reader.rb:14]: Why are you setting Request ID to Tracking ID?

Comment (RaVbaker) [reply lib/kiev/context_reader.rb:14]: to have consistent and same value in both fields always. With priority to tracking_id.

File: test/rails_integration_test.rb [hunk 2]
```diff
@@ -26,29 +27,32 @@ def test_simple_get
...
-        {},
-        "X-Request-Id" => "external-uu-rid",
-        "X-Request-Depth" => "0",
-        "X-Tree-Path" => "AA"
+        params: {},
+        headers: {
+          "X-Request-Id" => "external-uu-rid",
+          "X-Request-Depth" => "0",
+          "X-Tree-Path" => "AA"
+        }
```
Comment (bl-rodrigo) [inline test/rails_integration_test.rb:35]: Would be nice to add a test for the tracking_id with cases like both headers come on the request the tracking id should have the priority.

**Review by bl-rodrigo — COMMENTED** [review]
LGTM. I believe if we just need to add one test case scenario and it is ready to go.

File: test/rails_integration_test.rb [hunk 3]
```diff
@@ -8,6 +8,7 @@ class RailsIntegrationTest < ActionDispatch::IntegrationTest
 
     def test_simple_get
       get("/")
+      p log_first
```
[the remark this answers is not in the document]
Comment (RaVbaker) [reply test/rails_integration_test.rb:11]: nice catch 🦅 👁️

File: test/rails_integration_test.rb (hunk 2 above)
Comment (RaVbaker) [reply test/rails_integration_test.rb:35]: @bl-rodrigo updated

**Review by giulliano-bueno — APPROVED** [review]

**Pull request merged**